x/gateway/keeper/msg_server_stake_gateway.go

```diff
@@ -24,46 +26,72 @@ func (k msgServer) StakeGateway(
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
 	logger := k.Logger().With("method", "StakeGateway")
-	logger.Info(fmt.Sprintf("About to stake gateway with msg: %v", msg))
+	logger.Info(fmt.Sprintf("about to stake gateway with msg: %v", msg))
 
 	if err := msg.ValidateBasic(); err != nil {
-		return nil, err
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	// Retrieve the address of the gateway
+	gatewayAddress, err := sdk.AccAddressFromBech32(msg.Address)
+	// NB: This SHOULD NEVER happen because msg.ValidateBasic() validates the address as bech32.
+	if err != nil {
+		// TODO_TECHDEBT(#384): determine whether to continue using cosmos logger for debug level.
+		logger.Info(fmt.Sprintf("could not parse address %q", msg.Address))
+		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
 	// Check if the gateway already exists or not
-	var err error
-	var coinsToEscrow sdk.Coin
+	var (
+		coinsToEscrow sdk.Coin
+		totalStake    sdk.Coin
+	)
 	gateway, isGatewayFound := k.GetGateway(ctx, msg.Address)
 	if !isGatewayFound {
-		logger.Info(fmt.Sprintf("Gateway not found. Creating new gateway for address %q", msg.Address))
+		logger.Info(fmt.Sprintf("gateway not found. Creating new gateway for address %q", msg.Address))
 		gateway = k.createGateway(ctx, msg)
 		coinsToEscrow = *msg.Stake
+		totalStake = *msg.Stake
 	} else {
-		logger.Info(fmt.Sprintf("Gateway found. About to try and update gateway for address %q", msg.Address))
+		logger.Info(fmt.Sprintf("gateway found. About to try and update gateway for address %q", msg.Address))
 		currGatewayStake := *gateway.Stake
 		if err = k.updateGateway(ctx, &gateway, msg); err != nil {
 			logger.Error(fmt.Sprintf("could not update gateway for address %q due to error %v", msg.Address, err))
-			return nil, err
+			return nil, status.Error(codes.InvalidArgument, err.Error())
 		}
 		coinsToEscrow, err = (*msg.Stake).SafeSub(currGatewayStake)
 		if err != nil {
-			return nil, err
+			return nil, status.Error(
+				codes.InvalidArgument,
+				types.ErrGatewayInvalidStake.Wrapf(
+					"stake (%s) must be higher than previous stake (%s)",
+					msg.Stake, currGatewayStake,
+				).Error(),
+			)
 		}
-		logger.Info(fmt.Sprintf("Gateway is going to escrow an additional %+v coins", coinsToEscrow))
+		totalStake = currGatewayStake.Add(*msg.Stake)
```

Comment:
`msg.Stake` is the new total stake, we should not add it to the current stake.
```
newStake == msg.Stake
i.e.
newStake == oldStake + coinsToEscrow
```

Reply:
:facepalm: Thank you for catching this.